src/app/applications/components/application-details/application-details.tsx

```diff
@@ -342,7 +336,7 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{ na
     }
 
     private async deleteApplication() {
-        AppUtils.deleteApplication(this.props.match.params.name, this.appContext, () => {
+        await AppUtils.deleteApplication(this.props.match.params.name, this.appContext).then(() => {
```

Comment:
No need to use await/async and then at the same time. Please use either 
```
AppUtils.deleteApplication(this.props.match.params.name, this.appContext).then(() => {
  this.appContext.apis.navigation.goto('/applications');
});
```

or 

```
await AppUtils.deleteApplication(this.props.match.params.name, this.appContext); 
this.appContext.apis.navigation.goto('/applications');
```